src/interface/container_iterator.hpp

```diff
@@ -38,9 +128,8 @@ class ContainerIterator {
 
   /// initializes the iterator with a container and a flag to match
   /// @param c the container on which you want the iterator
```

Comment:
Should we remove class ContainerIterator, the old iterator mechanism?

Reply:
I vote yes. But we should be careful to integrate the new mechanism everywhere where it's used in the code. Breaking downstream is probably fine. Breaking code in parthenon itself is probably not a good idea.

Reply:
OK, it turns out we can't eliminate the class because it satisfies a need not filled by the VariablePack class.  If you look at src/interface/update.cpp, the code accesses the flux components of the variable, which are not provided by the VariablePack class.

Reply:
Agreed about the flux stuff.  I spent a little time earlier thinking about that.  No solution coded up just yet though.

Reply:
VariablePack could also have a fluxPack member that is a pack of the fluxes in the same order.  Problem occurs for variables that don't have a flux array.

Reply:
If a variable doesn't have a flux array, it can be not added to the fluxpack? Or it can be added to the fluxpack as an empty view?

Reply:
Probably an empty view since we want to keep the indexing same for fluxes and the variable.

Reply:
we might need logic for checking if a view is nonempty in relevant bits of code then. Not that that's an issue.

Reply:
Maybe an assert for debugging but otherwise I don't think we do because the flux is only accessed by physics when there is one and the physics will be accessing variables by named indices.